Sources/StreamChat/Repositories/AuthenticationRepository.swift

```diff
@@ -135,8 +149,23 @@ class AuthenticationRepository {
     ///   - userInfo:       The user information that will be created OR updated if it exists.
     ///   - tokenProvider:  The block to be used to get a token.
     func connectUser(userInfo: UserInfo?, tokenProvider: @escaping TokenProvider, completion: @escaping (Error?) -> Void) {
```

Comment:
The biggest problem I see now is that we heavily depend on UserInfo to be filled. Otherwise, we will be constantly logging out on every session.

That's why the previous approach was using the newly fetched token to determine if things were the same or not 🙃

Reply:
UserInfo is always passed when logging a regular user. New logic made it evident